src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7785,5 +7785,26 @@ static bool Dummy(bool x, object y, object z)
 1
 2");
         }
+
+        [Fact]
+        public void ThrowName()
+        {
+            var source =
+@"
+class Program
+{
+    static void Main(string[] args)
+    {
+        var y = null ?? throw null;
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
+            compilation.VerifyDiagnostics(
+                // (6,17): error CS0019: Operator '??' cannot be applied to operands of type '<null>' and '<throw>'
```

Comment:
Yes, the same for <null>, method groups, lambdas, etc. It is the compiler's existing style.